esmvalcore/_recipe.py

```diff
@@ -845,6 +899,16 @@ def append(group_prefix, var):
     return derive_input
 
 
+def _remove_time_preproc(fxprofile):
+    """Remove all time preprocessors from the fx profile."""
+    fxprofile = deepcopy(fxprofile)
+    for key in fxprofile:
+        if key in TIME_PREPROCESSORS:
+            fxprofile[key] = False
+
+    return fxprofile
+
+
 def _get_preprocessor_task(variables, profiles, config_user, task_name):
```

Comment:
90 lines and 7 levels of indentation in one function? Please refactor.

Reply:
K boss, refactored it by taking the fx ancestry up into a separate function so we can easily remove when we get rid of ancestry -> @bouweandela

Reply:
done :beer: